DDD-4.md

```diff
@@ -0,0 +1,168 @@
+  // See Incremental Snapshots for common incremental-snapshot key/value pairs 
+}
+```
+
+## Transaction Monitoring
+
+A subset of Debezium connectors manage transaction-based offset attributes.
+If a connector supports transaction monitoring features, the following additional offset attributes are managed by the connector:
+
+```json
+{
+  "transaction_id": "<transaction_id>", // string - the current transaction being processed
+  "transaction_data_collection_order_<table-name>": 1 // long
+}
+```
+
+The `transaction_data_collection_order_<table-name>` entry is a key that is dynamic where the `<table-name>` is replaced by the fully qualified `TableId`.
+This means there can be multiple entries for different tables in the offsets where the value associated with each key represents the number of events observed within the transaction for that specific table.
+
+## Incremental Snapshots
+
+A subset of Debezium connectors manage incremental snapshot offset attributes.
+If a connector supports Incremental Snapshots, these attributes will only be visible in the offsets when an Incremental Snapshot is in progress.
+These attributes will not appear if the incremental snapshot has concluded or is never started.
+The basic format of incremental snapshot attributes is:
+
+```json
+{
+  "incremental_snapshot_primary_key": "<data>", // string, Object[] serialized to byte array converted to Hex
+  "incremental_snapshot_maximum_key": "<data>", // string, Object[] serialized to byte array converted to Hex
+  "incremental_snapshot_collections": "<json>" // string
+}
+```
+The incremental snapshot context will serialize the collections that are still to be incrementally snapshot to the connector offsets as JSON and store this serialized value in the `incremental_snapshot_collections` field.
+This field is a JSON array of entries that have the following format:
+
+```json
+{
+  "incremental_snapshot_collections_id": "<database>.<schema>.<table>", // string, the TableId
+  "incremental_snapshot_collections_additional_condication": "ID = 25", // string, the additional where condition, may be null
+  "incremental_snapshot_collection_surrogate_key": "NAME" // string, may be null.
+}
+```
+
+Therefore, the expected layout of the `incremental_snapshot_collections` field would be `[{obj1}, {obj2}, ...]`.
+
+## Proposed Solution
+
+Provide a signal action to read and write offsets for each connector. The signal action will be a new action type in the Debezium signal framework.
+
+### Signal Action
+
+The `Change Offset` signal action will be used to change the offset of a connector. The action will have the following attributes:
+
+* `name` - The name of the action. This will be `change-offset`.
+* `offset-position` - The new offset position to set (which corresponds to `lsn_commit` in the PostgreSQL connector).
+* `last-offset-position` - The last offset position to set (which corresponds to `lsn_proc` in the PostgreSQL connector).
```

Comment:
I think we need to come up with better naming and descrption/explanation, current one is a bit confusing.
Also I think we need an example of the signal message in the DDD too.